tests/execution/test-list-argument.md

```diff
@@ -0,0 +1,32 @@
+# With List args
+
+```graphql @config
+schema @server(queryValidation: true) @upstream(baseURL: "http://localhost:3000") {
+  query: Query
+}
+
+type Query {
+  f1(q: [Int!]!): T1 @http(path: "/api", query: [{key: "q", value: "{{.args.q}}"}])
+}
+
+type T1 {
+  numbers: [Int]
+}
+```
+
+```yml @mock
+- request:
+    method: GET
+    url: http://localhost:3000/api?q=1,2,3
```

Comment:
There are multiple ways to encode this — 

```curl
# Option 1
/api?q=1,2,3

# Option 2
/api?q=1&q=2&q=3

# Option 3
/api?q[]=1&q[]=2&q[]=3
```

I think default should be option 2, but the user should be able to decide which option to use for encoding lists in the url. 

This setting should be specified in `upstream` however it can be overridden in `@http` if required.

Reply:
The main problem that I see here, is that this way of handling lists is very different from how we handle lists in Bulk APIs.